src/OpenCensus.Collector.Implementation.Common/ListenerHandler.cs

```diff
@@ -16,16 +16,17 @@
 
 namespace OpenCensus.Collector.Implementation.Common
 {
+    using System;
     using System.Diagnostics;
     using OpenCensus.Trace;
 
-    internal class ListenerHandler
+    internal abstract class ListenerHandler
```

Comment:
I feel it may be better to simply have two copies of implementation instead of abstracting it away. So basically copu/paste code.